docs/conch/examples/sshsimpleclient.py

```diff
@@ -66,20 +67,32 @@ def getGenericAnswers(self, name, instruction, questions):
         return defer.succeed(answers)
 
     def getPublicKey(self):
+        args = parser.parse_args()
+        public_key_path = (
+            args.client_public_key_path
+            if args.client_public_key_path
+            else CLIENT_RSA_PUBLIC
+        )
         if (
-            not CLIENT_RSA_PUBLIC
-            or not os.path.exists(CLIENT_RSA_PUBLIC)
+            not public_key_path
+            or not os.path.exists(public_key_path)
             or self.lastPublicKey
         ):
             # the file doesn't exist, or we've tried a public key
             return
-        return keys.Key.fromFile(filename=CLIENT_RSA_PUBLIC)
+        return keys.Key.fromFile(filename=public_key_path)
 
     def getPrivateKey(self):
         """
         A deferred can also be returned.
         """
-        return defer.succeed(keys.Key.fromFile(CLIENT_RSA_PRIVATE))
+        args = parser.parse_args()
```

Comment:
This is deeply weird... you're referencing a `parser` that's conditionally defined in a global scope. And doing that twice?

I don't think that these are appropriate changes to make, as this example deliberately hard-codes things to keep the boilerplate to a minimum.

Reply:
Hey! Could you be more specific about `parser` providing me some example? 

Regarding the hard-coded values: maybe having more flexible examples could be better also for twisted devs, we discussed that with @adiroiban https://github.com/twisted/twisted/pull/12002#issuecomment-1971109313

Reply:
Thank you for providing that context! I'd only read #12210 before reviewing.

Are you aware that you can run a server with `python -m twisted conch`? And that the `conch` script that ships with Twisted is an SSH client?

<details>
<summary>Answers to your questions, but probably irrelevant</summary>

You should parse the parameters in a single location (I suggest defining a `main` function) and thread the values in through `SimpleTransport` down to `SimpleUserAuth`. To do so you'd have to add a `__init__` methods to both `SimpleTransport` and `SimpleUserAuth` that accept those parameters and store them as instance variables. This will add some boilerplate which is presumably why the examples weren't written that way to begin with. Don't rely on global variables.

Unfortunately, these examples are a bit long in the tooth &mdash; `simplesshclient.py` is using `ClientCreator` when it should be using `HostnameEndpoint`. It calls `reactor.run()` when it should use `twisted.internet.task.react()`, etc. You don't have to address that in this PR, but I'd welcome those changes here or in a separate PR. (Really, we need to go through all of the examples and update them.)

Twisted also contains an argument parsing system: `twisted.web.usage.Options`. (I haven't done the archaeology, but `Options` surely predates `optparse` and `argparse` in the stdlib). You could use that, or stick with `argparse` &mdash; I'm indifferent &mdash; but you may be able to find examples if you search for `Options`.

</details>

Reply:
@twm thank you for the clarification! I will update this PR with the information you gave me, I ignored the fact that we can run a server using the command you mentioned.

Moreover, for the comments above, I think I should delete this PR and start it again from the trunk branch. 

At this point it makes sense to wait for #12300

Reply:
I was not aware of the `twisted couch` app.

I am always using the example files so that I can get an idea of what parts are needed and to allow to customize the ssh server.

I guess there are 2 use-cases here. 

* Case A - You just want to get an SSH server up and running for shell access and don't care how it works
* Case B - You want to setup a custom SSH server

For Case A, the `twisted conch` can help... but if you just need a simple/standard SSH server, I am not sure why you would bother with `conch` in the first place :)

For Case B, I think that the example file is much better, especially since there is not much narative documentation for conch ssh.

Reply:
> For Case A, the `twisted conch` can help... but if you just need a simple/standard SSH server, I am not sure why you would bother with `conch` in the first place :)

Presumably if someone is working with Conch they will want to try its "out of the box" behavior with the included command line, and it should work properly.  Ideally it should be more configurable so it addresses more of the Case B audience as well, in the same way that e.g. `twist web` can be used to deploy production services